modules/lava.lua

```diff
@@ -155,16 +158,69 @@ local function applyConfig(lavaConfig)
 end
 
 -- Generates a lava tide rhythm based on the spring modoptions.
+local function validateTideRhythm(modoptionDataRaw)
+	if modoptionDataRaw == nil or string.len(modoptionDataRaw) == 0 then
+		return false
+	end
+	modoptionDataTrim = modoptionDataRaw:gsub("%s+", "")
+
+	local advancedRhythm = {}
+	for tide in string.gmatch(modoptionDataTrim, "{(.-)}") do
+		local partRhythm = {}
+		for value in string.gmatch(tide, "[^,]+") do
+			if not tonumber(value) then
+				Spring.Echo("Lava Advanced Tide Rhythm data is not valid, non-number value: ", value)
+				return false
+			else 
+			table.insert(partRhythm, tonumber(value))
+			end
+		end
+		if #partRhythm ~= 3 then
+			Spring.Echo("Lava Advanced Tide Rhythm data is not valid, invalid tide definition: ", partRhythm)
+			return false
+		elseif not ((partRhythm[1] >= 0) and (partRhythm[2] > 0) and (partRhythm[3] > 0)) then
+			Spring.Echo("Lava Advanced Tide Rhythm data is not valid, negative or zero values: ", partRhythm)
+			return false
+		elseif (partRhythm[1] % 1 ~= 0) or (partRhythm[3]*gameSpeed % 1 ~= 0) then
+			Spring.Echo("Lava Advanced Tide Rhythm data is not valid, Height and Dwell frame must be a whole number: ", partRhythm)
```

Comment:
why? fractional height should be ok and dwell time is in seconds, so fractions are also ok (and roundable to frames if not perfect)